packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -70,175 +75,318 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
+            context,
+            "./src/worker/runner/javascript/context.js",
+            &[hash_util],
+        )?;
+        let ctx_import: Array<'_> = ctx_import.try_into().map_err(|err| {
             Error::FileImport(
-                "./src/worker/runner/javascript/context.js".into(),
-                "Couldn't get array (of functions) from 'context.js'".into(),
+                "./src/worker/runner/javascript/context.js".to_string(),
+                format!("Couldn't get array (of functions) from 'context.js': {err}"),
+            )
+        })?;
+        let experiment_ctx_prototype = ctx_import.get_index(scope, 0).ok_or_else(|| {
+            Error::V8(
+                "Couldn't get experiment_ctx_prototype from index 0 of the context.js return \
+                 values"
+                    .to_string(),
+            )
+        })?;
+        let sim_init_ctx_prototype = ctx_import.get_index(scope, 1).ok_or_else(|| {
+            Error::V8(
+                "Couldn't get sim_init_ctx_prototype from index 1 of the context.js return values"
+                    .to_string(),
+            )
+        })?;
+        let gen_ctx = ctx_import.get_index(scope, 2).ok_or_else(|| {
+            Error::V8(
+                "Couldn't get gen_ctx from index 2 of the context.js return values".to_string(),
             )
         })?;
-        let experiment_ctx_prototype = ctx_import.get(0)?;
-        let sim_init_ctx_prototype = ctx_import.get(1)?;
-        let gen_ctx = ctx_import.get(2)?;
-
-        let gen_state = import_file(mv8, "./src/worker/runner/javascript/state.js", vec![
-            &hash_util,
-        ])?;
-        let fns = import_file(mv8, "./src/worker/runner/javascript/runner.js", vec![
-            &arrow,
-            &batches_prototype,
-            &experiment_ctx_prototype,
-            &sim_init_ctx_prototype,
-            &gen_ctx,
-            &gen_state,
-        ])?;
-        let fns = fns.as_array().ok_or_else(|| {
+
+        let gen_state = import_file(
+            scope,
+            context,
+            "./src/worker/runner/javascript/state.js",
+            &[hash_util],
+        )?;
+        let fns = import_file(
+            scope,
+            context,
+            "./src/worker/runner/javascript/runner.js",
+            &[
+                arrow,
+                batches_prototype,
+                experiment_ctx_prototype,
+                sim_init_ctx_prototype,
+                gen_ctx,
+                gen_state,
+            ],
+        )?;
+        let fns: Array<'_> = fns.try_into().map_err(|err| {
             Error::FileImport(
                 "./src/worker/runner/javascript/runner.js".into(),
-                "Couldn't get array (of functions) from 'runner.js'".into(),
+                format!("Couldn't get array (of functions) from 'runner.js': {err}"),
             )
         })?;
-        Ok(Self {
+
+        let start_experiment: Function<'_> = get_function_from_array(scope, fns, 0)?;
```

Comment:
Rather than needing the `get_function_from_array` function, and all of these calls, could we not just `fns.into_iter().map.collect()` (or `map` a `0..6` and index) and destructure that into [`start_experiment`, `start_sim`, ...]`?

Reply:
Replaced with a loop in https://github.com/hashintel/hash/pull/414/commits/35ae9b9269e500eb3af5664c58d525863a6082f8

Reply:
Much better :)